src/validation/validation.cpp

```diff
@@ -2724,27 +2730,26 @@ void UpdateTip(CBlockIndex *pindexNew)
         CheckAndAlertUnknownVersionbits(chainParams, chainActive.Tip());
     }
 
-    if (IsNov152018Scheduled()) // Set the global variables based on the fork state of the NEXT block
+    // Set the global variables based on the fork state of the NEXT block
+    // Always allow overwite of enableCanonicalTxOrder but for regtest)
+    if (AreWeOnBCHChain())
     {
-        if (IsNov152018Enabled(chainParams.GetConsensus(), pindexNew))
+        if (chainParams.NetworkIDString() != "regtest")
         {
             enableCanonicalTxOrder = true;
         }
-        else
-        {
-            enableCanonicalTxOrder = false;
-        }
     }
-    if (IsSv2018Scheduled())
+
+    if (AreWeOnSVChain() && IsSv2018Activated(chainParams.GetConsensus(), chainActive.Tip()))
+    {
+        maxScriptOps = SV_MAX_OPS_PER_SCRIPT;
+        excessiveBlockSize = SV_EXCESSIVE_BLOCK_SIZE;
+    }
+    else
```

Comment:
Nit: Ignoring ptschip's comment below about correctness, it's possible to simplify and remove the else clause here by doing:

```
maxScriptOps = MAX_OPS_PER_SCRIPT;
excessiveBlockSize = DEFAULT_EXCESSIVE_BLOCK_SIZE;

if (AreWeOnSVChain()) {

	// override with sv stuff
}
```

Reply:
@ptschip was right there was a logic error in the prev code. In fact the EB change was meant to be executed only on the SV chain. see ea81d7a for the new code